lib/person.rb

```diff
@@ -0,0 +1,19 @@
+class Person
+  attr_reader :name, :interests, :supplies, :person
+  def initialize(person)
+    @person = person
+    @name = person[:name]
+    @interests = person[:interests]
+    @supplies = {}
+  end
+
+  def add_supply(supply, amount)
+    supplies[supply] = amount
+  end
+
+  def can_build?(craft)
```

Comment:
Rather than checking that the lists are entirely equal we want to check if a person has enough supplies.

```
craft.supplies_required.each do |supply, quantity_needed|
  if !@supplies[supply.to_s] || @supplies[supply.to_s] < quantity_needed
    return false
  end
end
true
```